vllm/entrypoints/openai/serving_completion.py

```diff
@@ -195,24 +180,30 @@ async def create_completion(
                 request_id,
                 created_time,
                 model_name,
-                num_prompts=len(prompts),
+                num_prompts=num_prompts,
                 tokenizer=tokenizer,
                 request_metadata=request_metadata)
 
         # Non-streaming response
-        final_res_batch: List[Optional[RequestOutput]] = [None] * len(prompts)
+        final_res_batch: List[Optional[RequestOutput]] = [None] * num_prompts
         try:
             async for i, res in result_generator:
                 final_res_batch[i] = res
+        except asyncio.CancelledError:
+            return self.create_error_response("Client disconnected")
+        except ValueError as e:
+            # TODO: Use a vllm-specific Validation Error
+            return self.create_error_response(str(e))
 
+        try:
```

Comment:
@DarkLight1337 apologies I didn't get a chance to review this last week .. ran into some things while trying to resolve conflicts with another pending PR :)

It looks like there are a few changes not directly related to chat embeddings.

Wondering the reason for splitting into two try/except blocks here in particular (I guess similar in serving_embedding.py)

Reply:
It is just to make clear that `asyncio.CancelledError` can only happen while iterating through the result generator.